lib/src/main/java/graphql/nadel/NextgenEngine.kt

```diff
@@ -265,13 +265,23 @@ internal class NextgenEngine(
         executionContext: NadelExecutionContext,
         hydrationDetails: ServiceExecutionHydrationDetails,
     ): ServiceExecutionResult {
-        return executeTopLevelField(
-            topLevelField = topLevelField,
-            service = service,
-            executionContext = executionContext.copy(
-                hydrationDetails = hydrationDetails,
-            ),
-        )
+        return try {
+            executeTopLevelField(
+                topLevelField = topLevelField,
+                service = service,
+                executionContext = executionContext.copy(
+                    hydrationDetails = hydrationDetails,
+                ),
+            )
+        } catch (e: Exception) {
+            when (e) {
+                is GraphQLError -> newServiceExecutionErrorResult(
+                    field = topLevelField,
+                    error = e,
+                )
+                else -> throw e
+            }
+        }
```

Comment:
@felipe-gdr I left the partitioned stuff untouched.

I think it's better to not handle it? The way I stumbled across this was I fed an invalid ARI into hydration, then we raised a `InvalidAriException`

I guess for partitioned stuff if there's an invalid ARI we should just stop the entire call?